### PR TITLE
Fixes to the CSV Export.

### PR DESCRIPTION
**Pull request by andrewtowe-rws**

### CODE REVIEW

File: cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java [hunk 1]
```diff
@@ -1064,17 +1079,20 @@ public boolean writeAllToCSV( IndexDefinition index, String query_string, List<S
...
+
+				}
 
 			}
-			while ( hits.size() != 0 ); // Zero hits mark the end of the scroll and the
+			while ( ! hits.isEmpty() ); // Zero hits mark the end of the scroll and the
 										// while
 										// loop.
-			return true;
+				CloudExecutionEnvironment.getSimpleCurrent().getSimpleSearch().closePointInTime(pitId);
```
Comment (preston-mccumber) [inline cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java:1094]: You don't have to used CloudExecutionEnvironment and ISearch to access closePointInTime. It's available locally.

**Review by preston-mccumber — APPROVED** [review]
Looks good. Only a couple of minor comments.

File: cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java [hunk 2]
```diff
@@ -1018,7 +1013,7 @@ public boolean reindex( IStorage storage, Kind... kinds )
 	}
 
 	@Override
-	public boolean writeAllToCSV( IndexDefinition index, String query_string, List<SearchFieldId> sorted_header, ICsvListWriter list_writer, CellProcessor[] cell_processors )
+	public boolean writeAllToCSV( IndexDefinition index, String query_string, List<SearchFieldId> sorted_header, FieldName id_field, ICsvListWriter list_writer, CellProcessor[] cell_processors )
```
[the remark this answers is not in the document]
Comment (andrewtowe-rws) [reply cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java:1016]: The id_field is used in finding the previous_id.

File: cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java (hunk 1 above)
Comment (andrewtowe-rws) [reply cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java:1094]: removed.

File: cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java [hunk 3]
```diff
@@ -1029,22 +1024,42 @@ public boolean writeAllToCSV( IndexDefinition index, String query_string, List<S
...
+						.query(new Query.Builder().queryString(new QueryStringQuery.Builder().query(query_string).build()).build())//
+						.from(0)//
+						.sort(new SortOptions.Builder().field(FieldSort.of(fs -> fs.field(FieldSortBuilder.DOC_FIELD_NAME).order(SortOrder.Asc))).build());
+
+				if ( previous_id != null )
+				{
+					search_request_builder.searchAfter(FieldValue.of(previous_timestamp), FieldValue.of(previous_id));
+				}
+				SearchRequest request = search_request_builder.build();
+				search_response = CloudExecutionEnvironment.getSimpleCurrent().getSimpleSearch().searchRaw(request);
```
[the remark this answers is not in the document]
Comment (andrewtowe-rws) [reply cloud/src/main/java/org/jimmutable/cloud/elasticsearch/ElasticSearchRESTClient.java:1051]: removed.

**Pull request merged**